rllib/core/learner/torch/torch_learner.py

```diff
@@ -84,46 +85,37 @@ def configure_optimizer_per_module(
     @override(Learner)
     def compute_gradients(
         self, loss: Union[TensorType, Mapping[str, Any]]
-    ) -> ParamDictType:
+    ) -> Tuple[ParamDictType, ResultDict]:
         for optim in self._optimizer_parameters:
             # set_to_none is a faster way to zero out the gradients
             optim.zero_grad(set_to_none=True)
         loss[self.TOTAL_LOSS_KEY].backward()
         grads = {pid: p.grad for pid, p in self._params.items()}
 
-        return grads
-
-    @OverrideToImplementCustomLogic_CallToSuperRecommended
-    @override(Learner)
-    def additional_update_per_module(
-        self, module_id: ModuleID, *, timestep: int, **kwargs
-    ) -> Mapping[str, Any]:
-        results = super().additional_update_per_module(module_id, timestep=timestep)
-
-        # Handle lr scheduling updates and apply new learning rates to the optimizers.
-        new_lr = self.lr_scheduler.update(module_id=module_id, timestep=timestep)
-        results.update({LEARNER_RESULTS_CURR_LR_KEY: new_lr})
-
-        return results
+        return grads, {}
 
     @override(Learner)
     def postprocess_gradients(
         self,
         gradients_dict: Mapping[str, Any],
-    ) -> Mapping[str, Any]:
+    ) -> Tuple[ParamDictType, ResultDict]:
         """Postprocesses gradients depending on the optimizer config."""
 
         # Perform gradient clipping, if necessary.
-        clip_gradients(
+        clip_by = self._optimizer_config.get("grad_clip_by")
+        global_norm = clip_gradients(
             gradients_dict,
             grad_clip=self._optimizer_config.get("grad_clip"),
-            grad_clip_by=self._optimizer_config.get("grad_clip_by"),
+            grad_clip_by=clip_by,
         )
+        stats = {}
+        if clip_by == "global_norm":
+            stats["gradients_global_norm"] = global_norm
 
-        return gradients_dict
+        return gradients_dict, stats
```

Comment:
I see why you need to return the stats. My proposal is that instead of returning stats and results dict in general (changing the API) we should define a better stats / logging API for the learner, in particular:

```

def __init__():
      self._metrics = ...

def record_metric(key, value):
      self._metrics[key] = value

def record_metric_dict(metric_dict):
      self._metrics.update(**metric_dict)
       
```

Then you can do:
```
self.record_metric("gradirent_global_norm", global_norm)
```

Reply:
That's a great idea! Would simplify things again a bit. I was also worried about the complex new return structure, but didn't see a way to avoid duplicate computations, e.g. for the global norm, so we do need some way of getting these metrics out of the different gradient methods.
I'll change this.

Reply:
done